internal/kafka/internal/config/gcp.go

```diff
@@ -0,0 +1,80 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/environments"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/go-playground/validator/v10"
+	"github.com/spf13/pflag"
+)
+
+const (
+	defaultGCPCredentialsFilePath = "secrets/gcp.api-credentials"
+
+	cloudProviderNameGCP = "gcp"
+)
+
+type GCPConfig struct {
+	GCPCredentials         GCPCredentials
+	gcpCredentialsFilePath string
+}
+
+type GCPCredentials struct {
```

Comment:
For now we make all attributes of the credentials required. We might relax this in the future if we find out some of them are not needed

Reply:
:+1:

Reply:
any other validations that would make sense? E.g. correct format of `ClientEmail` or `TokenURI`?

Reply:
I don't think so. If those are incorrect the provisioning will fail. Let's let OCM/GCP (the source of truth) deal with it.